### PR TITLE
chore: disable sealevel e2e when possible

### PR DESCRIPTION
**Pull request by daniel-savu**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -212,6 +212,15 @@ jobs:
         env:
           RUST_BACKTRACE: 'full'
 
+      - name: Check for Rust file changes
+        id: check-rust-changes
+        run: |
+          if [[ -n "$(git diff origin/${{ github.base_ref }} HEAD -- ./rust)" ]]; then
```
Comment (paulbalaji) [inline .github/workflows/test.yml:218]: this means even agent-config changes will lead to a full sealevel e2e, do we want that?

Comment (daniel-savu) [reply .github/workflows/test.yml]: it's hard not to have false positives but it's better than false negatives imo - cargo.toml and .lock files could change, and possibly other configs too

Comment (daniel-savu) [reply .github/workflows/test.yml]: since most agent config changes happen on new deploy batches, I think it's acceptable

Comment (paulbalaji) [reply .github/workflows/test.yml]: sgtm!

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -212,6 +212,15 @@ jobs:
         env:
           RUST_BACKTRACE: 'full'
 
+      - name: Check for Rust file changes
+        id: check-rust-changes
+        run: |
+          if [[ -n "$(git diff ${{ github.event.pull_request.base.sha }} HEAD -- ./rust)" ]]; then
```
Comment (paulbalaji) [inline .github/workflows/test.yml:218]:
```suggestion
          if [[ -n "$(git diff ${{ github.event.pull_request.head.sha || github.sha }} HEAD -- ./rust)" ]]; then
```

**Review by paulbalaji — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -212,6 +213,16 @@ jobs:
         env:
           RUST_BACKTRACE: 'full'
 
+      - name: Check for Rust file changes
+        id: check-rust-changes
+        run: |
+          if [[ -n "$(git diff ${{ github.event.pull_request.head.sha || github.sha }} ${{ github.event.pull_request.base.sha }} -- ./rust)" ]]; then
+            echo "rust_changes=true" >> $GITHUB_OUTPUT
+            echo "$(git diff ${{ github.event.pull_request.head.sha || github.sha }} ${{ github.event.pull_request.base.sha }} -- ./rust)"
```
Comment (paulbalaji) [inline .github/workflows/test.yml:221]:
```suggestion
          if [[ -n "$(git diff ${{ github.event.pull_request.head.sha || github.sha }} ${{ github.event.pull_request.base.sha }}~1 -- ./rust)" ]]; then
            echo "rust_changes=true" >> $GITHUB_OUTPUT
            echo "$(git diff ${{ github.event.pull_request.head.sha || github.sha }} ${{ github.event.pull_request.base.sha }}~1 -- ./rust)" 
```

**Pull request merged**